test/ui-test/src/screens/SettingsScreen.py

```diff
@@ -307,84 +312,95 @@ def set_bio(self, bio: str):
+        
+        click_obj_by_name(ProfileSettingsScreen.OPEN_SOCIAL_LINKS_DIALOG.value)
         click_obj_by_name(ProfileSettingsScreen.DISCORD_SOCIAL_LINK_IN_DIALOG.value)
-        verify(setText(ProfileSettingsScreen.DISCORD_SOCIAL_LINK_IN_DIALOG.value, discord), "set discord")
+        verify(setText(ProfileSettingsScreen.LINK_TEXT_FIELD.value, discord), "set discord")
+        click_obj_by_name(ProfileSettingsScreen.ADD_BUTTON.value)
+        
+        click_obj_by_name(ProfileSettingsScreen.OPEN_SOCIAL_LINKS_DIALOG.value)
         click_obj_by_name(ProfileSettingsScreen.TELEGRAM_SOCIAL_LINK_IN_DIALOG.value)
-        verify(setText(ProfileSettingsScreen.TELEGRAM_SOCIAL_LINK_IN_DIALOG.value, telegram), "set telegram")
+        verify(setText(ProfileSettingsScreen.LINK_TEXT_FIELD.value, telegram), "set telegram")
+        click_obj_by_name(ProfileSettingsScreen.ADD_BUTTON.value)
+        
+        click_obj_by_name(ProfileSettingsScreen.OPEN_SOCIAL_LINKS_DIALOG.value)
         click_obj_by_name(ProfileSettingsScreen.CUSTOM_LINK_IN_DIALOG.value)
-        verify(setText(ProfileSettingsScreen.CUSTOM_LINK_IN_DIALOG.value, custom_link_text), "set custom link name")
-        click_obj_by_name(ProfileSettingsScreen.CUSTOM_URL_IN_DIALOG.value)
-        verify(setText(ProfileSettingsScreen.CUSTOM_URL_IN_DIALOG.value, custom_link), "set custom link url")
-
-        click_obj_by_name(ProfileSettingsScreen.CLOSE_SOCIAL_LINKS_DIALOG.value)
+        verify(setText(ProfileSettingsScreen.LINK_TEXT_FIELD.value, custom_link_text), "set custom link name")
+        verify(setText(ProfileSettingsScreen.LINK_TEXT_FIELD.value, custom_link), "set custom link url")
+        click_obj_by_name(ProfileSettingsScreen.ADD_BUTTON.value)
+        
         click_obj_by_name(SettingsScreenComponents.SAVE_BUTTON.value)
 
-    def verify_social_links(self, table):
-
-        twitter = ""
-        personal_site = ""
-        github = ""
-        youtube = ""
-        discord = ""
-        telegram = ""
-        custom_link_text = ""
-        custom_link = ""
-
-        if table is not None:
-            verify_equals(8, len(table)) # Expecting 8 as social media link fields to verify
-            twitter = table[0][0]
-            personal_site = table[1][0]
-            github = table[2][0]
-            youtube = table[3][0]
-            discord = table[4][0]
-            telegram = table[5][0]
-            custom_link_text = table[6][0]
-            custom_link = table[7][0]
-
-        verify_text_matching(ProfileSettingsScreen.TWITTER_SOCIAL_LINK.value, twitter)
-        verify_text_matching(ProfileSettingsScreen.PERSONAL_SITE_SOCIAL_LINK.value, personal_site)
+    def get_social_links(self) -> dict:
+        links = {}
+        try:
+            hover(ProfileSettingsScreen.LINKS_LIST.value)
+            links_obj = wait_exists_and_get_obj(ProfileSettingsScreen.LINKS_LIST.value)
+        except (LookupError, AssertionError) as err:
+            test.log(str(err))
```

Comment:
We have the `test.log` in `SquishDriverVerification`.

Reply:
removed